vital/tests/test_mesh.cxx

```diff
@@ -0,0 +1,236 @@
+void compare_meshes(const kwiver::vital::mesh_sptr& first,
+                    const kwiver::vital::mesh_sptr& second)
+{
+  kwiver::vital::mesh_vertex_array<3>& first_vertices = dynamic_cast
+    <kwiver::vital::mesh_vertex_array<3>&>( first->vertices() );
+  kwiver::vital::mesh_vertex_array<3>& second_vertices = dynamic_cast
+    <kwiver::vital::mesh_vertex_array<3>&>( second->vertices() );
+
+  EXPECT_EQ( first_vertices.size(), second_vertices.size() );
+  for(unsigned int i=0; i<first_vertices.size(); i++)
+  {
+    EXPECT_EQ( first_vertices[i].size(), second_vertices[i].size() );
+    for(unsigned int j=0; j<first_vertices[i].size(); j++)
+    {
+      EXPECT_EQ( first_vertices[i][j], second_vertices[i][j] );
+    }
+  }
+
+  kwiver::vital::mesh_face_array first_faces = *std::make_shared
+    <kwiver::vital::mesh_face_array>(first->faces());
+  kwiver::vital::mesh_face_array second_faces = *std::make_shared
+    <kwiver::vital::mesh_face_array>(second->faces());
+
+  EXPECT_EQ( first_faces.size(), second_faces.size() );
+  for(unsigned int i=0; i<first_faces.size(); i++)
+  {
+    EXPECT_EQ( first_faces[i].size(), second_faces[i].size() );
+    for(unsigned int j=0; j<first_faces[i].size(); j++)
+    {
+      EXPECT_EQ( first_faces[i][j], second_faces[i][j] );
+    }
+  }
+}
+
+// ----------------------------------------------------------------------------
+TEST(mesh, group_names)
+{
+  kwiver::vital::mesh_sptr cube_mesh = kwiver::testing::cube_mesh( 1.0 );
+  std::shared_ptr<kwiver::vital::mesh_face_array> faces =
+    std::make_shared<kwiver::vital::mesh_face_array>(cube_mesh->faces());
+
+  for(unsigned int i=0; i<faces->size(); i++){
+    EXPECT_EQ( faces->group_name(i), "" );
+  }
+
+  EXPECT_EQ( faces->make_group( "testing name" ), faces->size() );
+  EXPECT_EQ( faces->make_group( "second testing name" ), 0 );
+
+  for(unsigned int i=0; i<faces->size(); i++){
+    EXPECT_EQ( faces->group_name(i), "testing name" );
+  }
+
+  EXPECT_EQ( faces->group_name( faces->size()+1 ), "" );
+
+  std::set<unsigned int> faces_set = faces->group_face_set( "testing name" );
+  for(unsigned int i=0; i<faces->size(); i++){
+    EXPECT_NE( faces_set.find( i ), faces_set.end() );
+  }
+}
+
+// ----------------------------------------------------------------------------
+TEST(mesh, append)
+{
+  kwiver::vital::mesh_sptr first_mesh = kwiver::testing::cube_mesh( 1.0 );
+  first_mesh->compute_face_normals();
+  std::shared_ptr<kwiver::vital::mesh_face_array> first_faces =
+    std::make_shared<kwiver::vital::mesh_face_array>(first_mesh->faces());
+  first_faces->make_group( "first name" );
+
+  kwiver::vital::mesh_sptr second_mesh = kwiver::testing::grid_mesh( 1, 1 );
+  second_mesh->compute_face_normals();
+  std::shared_ptr<kwiver::vital::mesh_face_array> second_faces =
+    std::make_shared<kwiver::vital::mesh_face_array>(second_mesh->faces());
+  second_faces->make_group( "second name" );
```

Comment:
Making groups on each face array isn't doing anything meaningful.  However, if you make a group on the first face array before merging in more faces, that's another good way to test the groups.  You should be able to construct `first_faces` such that it contains three groups that contain the faces originally found in `first_faces`, `second_faces`, and `third_faces`